sigstore-java/src/main/java/dev/sigstore/tuf/MetaStore.java

```diff
@@ -41,26 +39,13 @@ public interface MetaStore extends MetaReader {
   void writeMeta(String roleName, SignedTufMeta<? extends TufMeta> meta) throws IOException;
 
   /**
-   * Once you have ascertained that your root is trustworthy use this method to persist it to your
-   * local store. This will usually only be called with a root loaded statically from a bundled
-   * trusted root, or after the successful verification of an updated root from a mirror.
-   *
-   * @param root a root that has been proven trustworthy by the client
-   * @throws IOException since some implementations may persist the root to disk or over the network
-   *     we throw {@code IOException} in case of IO error.
-   * @see <a
-   *     href="https://theupdateframework.github.io/specification/latest/#detailed-client-workflow">5.3.8</a>
-   */
-  void writeRoot(Root root) throws IOException;
-
-  /**
-   * This clears out the snapshot and timestamp metadata from the store, as required when snapshot
-   * or timestamp verification keys have changed as a result of a root update.
+   * Generic method to remove meta, useful when keys rotated in root. Deletion is not optional,
+   * implementers must ensure meta is removed from the storage medium.
    *
    * @throws IOException implementations that read/write IO to clear the data may throw {@code
    *     IOException}
    * @see <a
    *     href="https://theupdateframework.github.io/specification/latest/#detailed-client-workflow">5.3.11</a>
    */
-  void clearMetaDueToKeyRotation() throws IOException;
+  void clearMeta(String role) throws IOException;
```

Comment:
I was following the spec here: https://github.com/theupdateframework/specification/blob/master/tuf-spec.md?plain=1#L1366

Why is this changing?

Reply:
it's not, the storage engine is just being directed by the Updater. So Updater is now telling it what to clear.  We were forcing some logic on the storage engine before?

Reply:
https://github.com/sigstore/sigstore-java/pull/840/files#diff-7e790d13f39bb712cf8caac24dda4082ab6b8acb38693cb900a80d268f918b6eR120

Reply:
oh I see.  yeah that makes sense